Improved color configuration

8ware opened the pull request:
Implemented #5 according to the proposed solution in my comment.

torque commented:
Looks good except you seem to have accidentally picked up `'chapter-marker-after': BG_PLACEHOLDER` in `settings.moon`. This doesn't match the `-background` suffix check so it won't work properly. Fix that and I'll be happy to merge.

torque commented on src/settings.moon:
```diff
@@ -70,7 +81,7 @@ settings = {
 	-- color of chapter marker before/after it has been passed. BGR hex.
 	'chapter-marker-before': '7A77F2'
-	'chapter-marker-after': '2D2D2D'
+	'chapter-marker-after': BG_PLACEHOLDER
```
You seem to have accidentally picked this up in find/replace. This doesn't match the `-background` suffix check, so I don't think it'll work properly.

8ware commented:
You were right, I didn't considered that case during the post-processing but I replaced it intentionally. For now, I fixed it by extending the (`else`)`if` condition to check also for `chapter-marker-after`. 

I think it is desirable to change the `chapter-after-marker` color as well to maintain consistency. I also thought about how the color for the `chapter-marker-before` could be derived from the foreground color. For example, by reversing the hex code to RGB: `79FC9E` &rarr; `9EFC79`.

torque commented:
This is an idea, but I don't think it'll work too well, for example, if someone's foreground color is `FFFFFF`, swapping the values just gives you `FFFFFF` again.

I think I'll merge this as-is and if you want to, we can discuss options for setting the chapter marker color somehow using the foreground and background settings.

8ware commented:
Ok, I thought a little bit more about that.

In the first instance I thing that the approach swapping the red and blue value is sufficient enough since most users will either leave the default settings unchanged or choose a (non-gray) color. But let's consider the border cases:

In case of a gray value (including `000000` and `FFFFFF`) we could use the inverse, e.g. `2D2D2D` → `D2D2D2`. But this would obviously not work for color values in the middle, e.g. `878787` → `787878`. I have no clue how to solve that, yet, except to print a warning and mention this in the README.

Also this approach won't work for values where the red and blue values are equal, e.g. `870087`. In that case we could exchange the red and blue value with the green one, resulting in `008700` (which is not a good combination, anyway).

Any other cases I missed?

I think we should not forget that this is just a matter of convenience. If we give some hints in the README I think we could get around an implementation that is off the point. Otherwise we'll end up in a (general) color scheme manager. At most we could provide some predefined color sets.

torque commented:
I've thought about it a bit and I think what I'll do is just make the chapter markers use the foreground color by default. There's no real reason that they really require a third color. This will negate the need to worry about generating a third color from either the foreground or background and, of course, the options will still be configurable.
